src/main/scala/lc/core/captcha.scala

```diff
@@ -50,24 +53,54 @@ class Captcha {
     token.asInstanceOf[Int]
   }
 
+  val supportedinputType = Config.getSupportedinputType
+  val supportedLevels = Config.getSupportedLevels
+  val supportedMedia = Config.getSupportedMedia
+
+  private def validateParam(param: Parameters): Boolean = {
+    if (
+      supportedLevels.contains(param.level) &&
+      supportedMedia.contains(param.media) &&
+      supportedinputType.contains(param.input_type)
+    )
+      return true
+    else
+      return false
+  }
+
   def getChallenge(param: Parameters): Id = {
     try {
-      val tokenPstmt = Statements.tlStmts.get.tokenPstmt
-      val rs = tokenPstmt.executeQuery()
-      val tokenOpt = if (rs.next()) {
-        Some(rs.getInt("token"))
+      val validParam = validateParam(param)
+      val result = if (validParam) {
+        val tokenPstmt = Statements.tlStmts.get.tokenPstmt
+        tokenPstmt.setString(1, param.level)
+        tokenPstmt.setString(2, param.media)
+        tokenPstmt.setString(3, param.input_type)
+        val rs = tokenPstmt.executeQuery()
+        val tokenOpt = if (rs.next()) {
+          Some(rs.getInt("token"))
+        } else {
+          None
+        }
+        val updateAttemptedPstmt = Statements.tlStmts.get.updateAttemptedPstmt
+        val token = tokenOpt.getOrElse(generateChallenge(param))
+        val uuidResult = if (token != -1) {
+          val uuid = getUUID(token)
+          updateAttemptedPstmt.setString(1, uuid)
+          updateAttemptedPstmt.executeUpdate()
+          uuid
+        } else {
+          "No Captcha for the provided parameters"
+        }
+        uuidResult
       } else {
-        None
+        "Invalid Parameters"
```

Comment:
Ideally, if something is wrong, it would be better to bubble the error up. If you want something more descriptive than `None`, you can use `Either[Error|Id]`.

Or you can create your own return type, something like:
```
sealed trait ChallengeResult
case class ChallengeError(msg: String) extends ChallengeResult
case class Challenge(id: ID) extends ChallengeResult

def getChallenge(param: Parameters): ChallengeResult = {
   ...
}
```

Reply:
Ah, perfect. Exactly what I needed.